flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaSinkITCase.java

```diff
@@ -123,18 +130,19 @@ public class KafkaSinkITCase extends TestLogger {
     private SharedReference<AtomicBoolean> failed;
     private SharedReference<AtomicLong> lastCheckpointedRecord;
 
-    @ClassRule
+    @Container
     public static final KafkaContainer KAFKA_CONTAINER =
             createKafkaContainer(KAFKA, LOG)
                     .withEmbeddedZookeeper()
                     .withNetwork(NETWORK)
                     .withNetworkAliases(INTER_CONTAINER_KAFKA_ALIAS);
 
-    @Rule public final SharedObjects sharedObjects = SharedObjects.create();
+    @RegisterExtension
+    public final SharedObjectsExtension sharedObjects = SharedObjectsExtension.create();
 
-    @Rule public final TemporaryFolder temp = new TemporaryFolder();
+    @TempDir public Path temp;
 
-    @BeforeClass
+    @BeforeAll
     public static void setupAdmin() {
```

Comment:
```suggestion
    static void setupAdmin() {
```
could be package private